File: 3206-alternating-groups-i/3206-alternating-groups-i.py
class Solution:
    def numberOfAlternatingGroups(self, colors: List[int]) -> int:
        n = len(colors)
        count = 0
        for i in range(n-2):
            if colors[i] == colors[i+2] and colors[i] != colors[i+1]:
                count += 1
        if colors[0] == colors[n-2] and colors[0] != colors[n-1]:
            count += 1
        if colors[1] == colors[n-1] and colors[1] != colors[0]:
            count += 1
        return count

File: LeetCode/3206-alternating-groups-i/3206-alternating-groups-i.java
class Solution {
    public int numberOfAlternatingGroups(int[] colors) {
        int n = colors.length;
        int count = 0;
        for (int i = 0; i < n - 2; i++) {
            if (colors[i] == colors[i+2] && colors[i] != colors[i+1]) {
                count++;
            }
        }
        if (colors[0] == colors[n-2] && colors[0] != colors[n-1]) {
            count++;
        }
        if (colors[1] == colors[n-1] && colors[1] != colors[0]) {
            count++;
        }
        return count;
    }
}